rocAL/rocAL/include/loaders/audio/audio_loader.h

```diff
@@ -0,0 +1,89 @@
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <string>
+#include <thread>
+#include <vector>
+#include "commons.h"
+#include "circular_buffer.h"
+#include "audio_read_and_decode.h"
+#include "meta_data_reader.h"
+//
+// AudioLoader runs an internal thread for loading an decoding of audios asynchronously
+// it uses a circular buffer to store decoded frames and audios for the user
+class AudioLoader : public LoaderModule {
+public:
+    explicit AudioLoader(void* dev_resources);
+    ~AudioLoader() override;
+    LoaderModuleStatus load_next() override;
+    void initialize(ReaderConfig reader_cfg, DecoderConfig decoder_cfg, RocalMemType mem_type, unsigned batch_size, bool keep_orig_size=false) override;
+    void set_output (rocalTensor* output_audio) override;
+    void set_random_bbox_data_reader(std::shared_ptr<RandomBBoxCrop_MetaDataReader> randombboxcrop_meta_data_reader) override {};
+    size_t remaining_count() override; // returns number of remaining items to be loaded
+    size_t last_batch_padded_size() override;
+    void reset() override; // Resets the loader to load from the beginning of the media
+    Timing timing() override;
+    void start_loading() override;
+    LoaderModuleStatus set_cpu_affinity(cpu_set_t cpu_mask);
+    LoaderModuleStatus set_cpu_sched_policy(struct sched_param sched_policy);
+    std::vector<std::string> get_id() override;
+    decoded_image_info get_decode_image_info() override;
+    crop_image_info get_crop_image_info() override;
+    void set_prefetch_queue_depth(size_t prefetch_queue_depth)  override;
+    void set_gpu_device_id(int device_id);
+    void shut_down() override;
+private:
+    bool is_out_of_data();
+    void de_init();
+    void stop_internal_thread();
+    std::shared_ptr<AudioReadAndDecode> _audio_loader;
+    LoaderModuleStatus update_output_audio();
+    LoaderModuleStatus load_routine();
+    rocalTensor* _output_tensor;
+    std::vector<std::string> _output_names;//!< audio name/ids that are stores in the _output_audio
+    size_t _output_mem_size;
+    MetaDataBatch* _meta_data = nullptr;//!< The output of the meta_data_graph,
+    std::vector<std::vector <float>> _bbox_coords;
+    bool _internal_thread_running;
+    size_t _batch_size;
+    std::thread _load_thread;
+    RocalMemType _mem_type;
+    decoded_image_info _decoded_img_info;
```

Comment:
Can we not have a separate structure for audio? 
decoded_audio_info _decoded_audio_info;

Reply:
That is not working out, we are getting issues